internal/devtools_browser.py

```diff
@@ -584,15 +589,16 @@ def run_lighthouse_test(self, task):
             command = ['lighthouse',
                        '"{0}"'.format(self.job['url']),
                        '--channel', 'wpt',
-                       '--disable-network-throttling',
-                       '--disable-cpu-throttling',
-                       '--throttling-method', 'provided',
                        '--enable-error-reporting',
                        '--max-wait-for-load', str(int(time_limit * 1000)),
                        '--port', str(task['port']),
                        '--output', 'html',
                        '--output', 'json',
                        '--output-path', '"{0}"'.format(output_path)]
+            if self.job['lighthouse_config_path']:
+                command.extend(['--config-path', self.job['lighthouse_config_path']])
```

Comment:
Do you need to do any sanitizing here or are you going to make absolutely sure that it is only ever statically defined in the server? Otherwise you can get arbitrary code execution on the agents.

Reply:
Ah, good point. Do you think a simple `os.access(path, os.R_OK)` would suffice?

Reply:
It looks like you got rid of the disable cpu and network throttling in the "else" case. Was that intentional?

As far as sanitizing, a couple of options come to mind:

1 - Store the configs on the server and pass the json down as part of the test. The agent could write the config to the temp path and there's no risk of execution. That also keeps you from having to build new agent images every time the config changes.

2 - Keep the configs in a well-known location and only let the test specify the name of the config file to use. Then you can make sure it's only alphanumeric characters.

Reply:
Do you mean the `--disable-{network,cpu}-throttling` flags? Those were removed a couple of versions ago in favour of `--throttling-method`.

I'll go with your first suggestion and have the full config JSON passed to the agent. This requires slightly more work for the test runner (can't just point at known LH configs) but agree it's way safer.